src/cpp/include/openvino/genai/whisper_pipeline.hpp

```diff
@@ -17,9 +17,10 @@ using OptionalWhisperGenerationConfig = std::optional<WhisperGenerationConfig>;
 
 using RawSpeechInput = std::vector<float>;
 
+class WhisperPipelineImplBase;
```

Comment:
could you please hide it inside private section of WhisperPipeline as it's done for Text2Image ?

https://github.com/openvinotoolkit/openvino.genai/blob/abbc695715a1d0a92facd807fe63c6d2088b99e7/src/cpp/include/openvino/genai/text2image/pipeline.hpp#L144-L151

Reply:
I don't think so, it will prohibit inheritance from this class. I believe this is the reason why it's done the same way for `LLMPipelineImplBase`:
https://github.com/openvinotoolkit/openvino.genai/blob/abbc695715a1d0a92facd807fe63c6d2088b99e7/src/cpp/include/openvino/genai/llm_pipeline.hpp#L99-L115

Reply:
see how it's done in Text2Image

Inherited class should be a friend of main public class.

Reply:
ping